common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/SwitchBrowserUriHelper.kt

```diff
@@ -105,21 +106,71 @@ object SwitchBrowserUriHelper {
+     *
+     * @param actionUri The action uri to be opened.
+     * @param redirectUrl The redirect url to be included in the resume uri.
+     * @param clientId The client id to be included in the resume uri.
+     *
+     * @return The resume uri constructed from the bundle.
+     * e.g. actionUri?client_id=client-id&redirect_uri=redirect-uri
+     */
+    fun buildResumeUri(actionUri: String, redirectUrl: String, clientId: String): Uri {
+        // Query parameters for the resume uri.
+        val queryParams = hashMapOf<String, String>()
+        queryParams[OAuth2.CLIENT_ID] = clientId
+        queryParams[OAuth2.REDIRECT_URI] = redirectUrl
+        // Construct the uri to the resume endpoint.
+        return buildSwitchBrowserUri(actionUri, queryParams)
+    }
+
+    /**
+     * Build the request headers for the resume endpoint.
+     *
+     * @param code The switch browser code.
+     *
+     * @return The request headers constructed from the bundle.
+     */
+    fun buildResumeRequestHeaders(code: String): HashMap<String, String> {
+        val requestHeaders = hashMapOf(CHALLENGE_RESPONSE_HEADER to code)
+        return requestHeaders
+    }
+
+    /**
+     * Check if the request is to start the switch the browser flow.
      *
      * The request is considered "switch_browser" if the URL
      * starts with the following pattern: {redirectUrl}/switch_browser
      *
      *
      * @param url The URL to be checked.
      * @param redirectUrl The redirect URL to be checked against.
-     * @return True if the request contains the required parameters, false otherwise.
+     * @return True if the request matches the pattern, false otherwise.
+     */
+    fun isStartSwitchBrowserRequest(url: String?, redirectUrl: String): Boolean {
+        val switchBrowserRedirectUrl = "${redirectUrl}/${SWITCH_BROWSER.START_PATH}"
+        if (url == null) {
+            return false
+        }
+        return url.startsWith(switchBrowserRedirectUrl, ignoreCase = true)
+    }
+
+    /**
+     * Check if the request is to switch the browser.
+     *
+     * The request is considered "switch_browser" if the URL
+     * starts with the following pattern: {redirectUrl}/switch_browser_resume
+     *
+     *
+     * @param url The URL to be checked.
+     * @param redirectUrl The redirect URL to be checked against.
+     * @return True if the request matches the pattern, false otherwise.
      */
-    fun isSwitchBrowserRequest(url: String?, redirectUrl: String): Boolean {
-        val switchBrowserRedirectUrl = "${redirectUrl}/${SWITCH_BROWSER.PATH}"
+    fun isResumeSwitchBrowserRequest(url: String?, redirectUrl: String): Boolean {
+        val switchBrowserRedirectUrl = "${redirectUrl}/${SWITCH_BROWSER.RESUME_PATH}"
         if (url == null) {
             return false
         }
-        return url.startsWith(switchBrowserRedirectUrl)
+        return url.startsWith(switchBrowserRedirectUrl, ignoreCase = true)
```

Comment:
Pretty much same code as on lines 151-155. Perhaps you can create a helper method as follows:

```java
fun isSwitchBrowserRequest(url: String?, redirectUrl: String, path: String) {
        val switchBrowserRedirectUrl = "${redirectUrl}/${path}"
        if (url == null) {
            return false
        }
        return url.startsWith(switchBrowserRedirectUrl, ignoreCase = true)
}

fun isResumeSwitchBrowserRequest(url: String?, redirectUrl: String): Boolean {
     return IsSwitchBrowserRequest(url, redirectUrl, SWITCH_BROWSER.RESUME_PATH)
}
```

Reply:
done